### DISCUSSION TITLE
Problema Connessione Caddy/DuckDNS

### DISCUSSION THREAD

**Discussion by Asdragan** [indomus/forum#181] [Networking]
Buonasera.

A seguito di un salto di corrente (credo, non so se in realtà sia dovuto ad altro, nel WE uso Home Assistant solo da casa e quindi in locale), non riesco più ad accedere da remoto a HA se non con un workaround (In pratica accedo in locale tramite VPN).

Il log Docker di DuckDNS mi da delle letture strane, ma in teoria funzionucchia 

[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 40-config: executing...
Retrieving subdomain and token from the environment variables
log will be output to docker log
Your IP was updated at Mon Apr 19 17:21:33 CEST 2021
[cont-init.d] 40-config: exited 0.
[cont-init.d] 99-custom-files: executing...
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-files: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
Your IP was updated at Mon Apr 19 17:25:07 CEST 2021
Your IP was updated at Mon Apr 19 17:30:08 CEST 2021
Something went wrong, please check your settings  Mon Apr 19 17:35:02 CEST 2021
Something went wrong, please check your settings  Mon Apr 19 17:40:02 CEST 2021
Something went wrong, please check your settings  Mon Apr 19 17:45:02 CEST 2021
Something went wrong, please check your settings  Mon Apr 19 17:50:02 CEST 2021
Something went wrong, please check your settings  Mon Apr 19 17:55:02 CEST 2021
Your IP was updated at Mon Apr 19 18:00:09 CEST 2021

Non aggiorna sempre l'IP, ma lo fa. Ho provato a rimuovere il Dock e rimetterlo, il comportamento non cambia.

Il Log di Caddy mi da invece questo errore. Ovviamente gli xxxxxx puntano al mio reale dominio. 

{"level":"error","ts":1618847701.3782465,"logger":"tls","msg":"job failed","error":"[xxxxxx.duckdns.org] unable to acquire lock 'issue_cert_xxxxx.duckdns.org': decoding lockfile contents: EOF"}

Anche in questo caso ho provato a rimuovere il dock e ripartire, ma non ho avuto successo. 

Il problema deve trovarsi da qualche parte nel mezzo fra i due dock, visto che per il resto caddy funziona (la VPN che uso infatti passa da lì con Wireguard).

**Comment by snakuzzo** [comment]
Cosa sarebbe il "Dock "? Forse volevi dire il container...
Non c'entra niente il container docker di Duckdns con quello di Caddy.
Tra i due container docker (e non dock) non c'è niente.

Prova a far ripartire Caddy solo dopo esserti assicurato di aver aggiornato l'associazione IP-FQDN. Magari prova ad eliminare i file sotto la dir dei dati di Caddy

**Comment by gualo1983** [comment]
Buona sera,
oggi ho riavviato il raspberry 3b+ e ho avviato i vari container dopo averli aggiornati. 
tutti i container sono partiti tranquillamente a parte caddy.
leggendo i log ho notato che non è riuscito a reperire i certificati da let's encryp e ho notato che ha provato a scaricarli da zerossl(non so manco cosa sia, immagino un alternativa a let's encryp).
a qualcuno è successo lo stesso? come posso risolvere?
